scripts/tuple_tree_generator/tuple_tree_generator/generators/typescript.py

```diff
@@ -0,0 +1,191 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import re
+from collections import namedtuple
+from typing import Optional
+from jinja2 import Environment
+from jinja2.utils import Markup
+
+from .jinja_utils import loader
+
+from ..schema import Schema
+from ..schema import (
+    StructDefinition,
+    StructField,
+    SimpleStructField,
+    SequenceStructField,
+    ReferenceStructField,
+)
+
+int_re = re.compile(r"(u)?int(8|16|32|64)_t")
+CPP_TYPE = re.compile(r"^((?P<container>[^<>]+<(?P<contained>.*)>)|(?P<simple>[^<>]*))$")
```

Comment:
What's this?

Reply:
>  a field might contain UpcastablePointer<model::Type> but I need just Type

Where does this happen? We should have got rid of those.

Reply:
I've refactored the `schema` package a bit, the regex is gone